doc/source/Tutorials/Scripting/CreatingConfigurationFiles3/index.md

```diff
@@ -0,0 +1,166 @@
+# Tutorial: Startup Configs, Part 3 # 
+
+In this final part of the startup config tutorial, we will demonstrate a startup config that adds a custom entry to the node menu.
+
+![A custom entry in the node menu](images/tutorialNodeMenuCustomEntry.png "A custom entry in the node menu")
+
+The entry will insert a Reference node and load a reference script into it. The sub-graph inside the Reference node consists of an OSLCode node that procedurally generates a texture of a Macbeth chart. This "custom" node should come in handy for lookdev shader development, and, as a bonus, provides a simple demonstration of procedural texture generation in OSL.
+
+Functionally, this node will be like a new node that came from a custom module. However, since the node is basically just an in-graph wrapper made of standard nodes, it will remain portable between Gaffer deployments, meaning that if it is saved inside a script, other users will not need to install additional modules in order to open that script.
+
+We hope that this simple config will be a springboard for adding further entries to the node menu. This solution is very powerful, as it allows you to customize existing nodes or insert References to your Gaffer deployment, without needing to code or compile Gaffer modules.
+
+
+## Startup Config #3: Custom Node Menu Entry ##
+
+As with the other startup configs, this one will run in the GUI app. Copy this code to a new a `customNodes.py` file in `~/gaffer/startup/gui`:
+
+```eval_rst
+.. code-block:: python
+    :linenos:
+
+    import Gaffer
+    import GafferUI
+    import IECore
+    import os
+
+    def MacbethTexture( menu ) :
+
+        graphEditor = menu.ancestor( GafferUI.GraphEditor )
```

Comment:
I find this approach really odd... Adding a node to the menu should be a lot simpler with no need to locate UI elements.

```
def __macbethTexture() :

	return Gaffer.Reference( "MacbethTexture" )

def __macbethTexutrePostCreator( node, menu ) :

	node.load( os.path.expandvars( "$GAFFER_ROOT/resources/references/macbethTexture.grf" ) )

nodeMenu = GafferUI.NodeMenu.acquire( application )
nodeMenu.append( "/Custom/MacbethTexture", __macbethTexture, postCreator=__macbethTexutrePostCreator, searchText = "MacbethTexture" )
```

The advantage of this approach (in addition to the code being simpler) is that the new node gets all the same treatment as a normal node (undoable, auto-placed, user defaults applied, etc).

You'll need to add some copy explaining what a postCreator is and why we need one in this case. Also note I made the functions private and lowerCamelCase (just a convention).

Reply:
This changes quite a lot of the remainder of this tutorial, so I haven't reviewed the rest of this one yet.

Reply:
Oh, my comment was misleading, you were already using `nodeMenu.append` so did already get most of those advantages... but I still think this new code is simpler / less scary since we don't need to know about graph editors and gadgets and whatnot.

Reply:
Ah, good. Cleaner/shorter code is much easier to write about. :)

Regarding the `postCreator`, is there anything special the user needs to know about it? Limitations or hidden features?

> private and lowerCamelCase

Happy to do, as long as I can use your prescription as an argument for using it everywhere else. :)

Reply:
I guess just that postCreator needs to be a callable with a particular signature `fn( node, menu )` and that it will be called after the new node has been added to the script.